i6_models/parts/conformer/mhsa_rel_pos.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+
+__all__ = ["ConformerMHSARelPosV1", "ConformerMHSARelPosV1Config"]
+
+from dataclasses import dataclass
+import math
+from typing import Optional
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+from i6_models.config import ModelConfiguration
+from i6_models.util import compat
+
+
+@dataclass
+class ConformerMHSARelPosV1Config(ModelConfiguration):
+    """
+    Attributes:
+        input_dim: input dim and total dimension for query/key and value projections, should be divisible by `num_att_heads`
+        num_att_heads: number of attention heads
+        att_weights_dropout: attention weights dropout
+        learnable_pos_emb: whether to use learnable relative positional embeddings instead of fixed sinusoidal ones
+        rel_pos_clip: maximal relative postion for embedding
+        with_pos_bias: whether to add additional position bias terms to the attention scores
+        pos_emb_dropout: dropout for the positional embeddings
+        dropout: multi-headed self attention output dropout
+        dropout_broadcast_axes: string of axes to which dropout is broadcast, e.g. "T" for broadcasting to the time axis
+                                setting to None to disable broadcasting
+    """
+
+    input_dim: int
+    num_att_heads: int
+    att_weights_dropout: float
+    dropout: float
+    learnable_pos_emb: bool = True
+    rel_pos_clip: Optional[int] = None
+    with_pos_bias: bool = False
+    pos_emb_dropout: float = 0.0
```

Comment:
You currently have those self-att related settings:

```
    num_att_heads: int  # num_heads in RF
    att_weights_dropout: float  # att_dropout in RF
    learnable_pos_emb: bool = True  # learnable_pos_emb in RF
    rel_pos_clip: Optional[int] = None  # learnable_pos_emb_clipping in RF
    with_pos_bias: bool = False  # with_pos_bias in RF
    pos_emb_dropout: float = 0.0  # pos_emb_dropout in RF
```

So, in RF, we have in addition those:
```
        with_bias: bool = True,
        with_linear_pos: bool = True,
        separate_pos_emb_per_head: bool = True,
```

`with_bias`: You implement it for True. But I have recently seen many people using False, e.g. Llama. So we might want to have this option.

`with_linear_pos`: I think you miss this. Why? This is different to ESPnet and I think also to how it is commonly done.

`separate_pos_emb_per_head`: You implement it for True. So maybe it's fine to not have this configurable now.

Reply:
`separate_pos_emb_per_head` was actually False in my code, now I added it.